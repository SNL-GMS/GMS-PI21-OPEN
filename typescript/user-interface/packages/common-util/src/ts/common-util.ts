import * as Timer from './common-util/performance-monitoring-util';

export * from './common-util/array-util';
export * from './common-util/compose-util';
export * from './common-util/delay-execution-util';
export * from './common-util/environment-util';
export * from './common-util/file-system-util';
export * from './common-util/formatting-util';
export * from './common-util/json-util';
export * from './common-util/logger';
export * from './common-util/min-max-util';
export * from './common-util/random-number-util';
export * from './common-util/range-util';
export * from './common-util/record-util';
export * from './common-util/sleep-util';
export * from './common-util/string-util';
export * from './common-util/time-util';
export * from './common-util/type-util';
export * from './common-util/uuid-util';
export * from './common-util/window-util';
export { KeyValue } from './types/key-value';
export { ValueType } from './types/value-type';
export { Timer };
