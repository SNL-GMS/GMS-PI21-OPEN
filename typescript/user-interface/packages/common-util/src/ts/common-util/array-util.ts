import filter from 'lodash/filter';
import includes from 'lodash/includes';
import uniq from 'lodash/uniq';

/**
 * @returns whether the two lists are unique
 */
export const isUnique = (a: unknown[] | undefined): boolean => a && uniq(a).length === a.length;

/** Gets a list the items that are duplicates */
export const getDuplicates = (arr: unknown[]): unknown[] =>
  filter(arr, (val, i: number, iteratee) => includes(iteratee, val, i + 1));
