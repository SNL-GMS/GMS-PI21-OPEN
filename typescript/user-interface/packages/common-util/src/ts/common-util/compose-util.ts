/**
 * Composes a function that invokes the given functions from right to left.
 *
 * @param funcs the functions to compose
 */
// TODO Replace this with lodash's flowRight function
// eslint-disable-next-line @typescript-eslint/ban-types, @typescript-eslint/explicit-module-boundary-types
export function compose(...funcs: Function[]) {
  funcs.reverse();
  return (...args: unknown[]) => {
    const [firstFunction, ...restFunctions] = funcs;
    // eslint-disable-next-line prefer-spread
    let result = firstFunction.apply(null, args);
    restFunctions.forEach(fnc => {
      result = fnc.call(null, result);
    });
    return result;
  };
}
