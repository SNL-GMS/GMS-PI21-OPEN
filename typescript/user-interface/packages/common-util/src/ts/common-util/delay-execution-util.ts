/**
 * Delay invoking a function for some number of milliseconds.
 *
 * @param func the function to call on a delayed time
 * @param delayMillis the number of milliseconds to delay
 */
export const delayExecution = async <T>(func: () => T, delayMillis = 50): Promise<T> =>
  new Promise<T>((resolve, reject) => {
    setTimeout(() => {
      try {
        resolve(func());
      } catch (e) {
        reject(e);
      }
    }, delayMillis);
  });

/**
 * Calls a function that accepts an arbitrary number of
 * arguments within setTimeout and returns clearTimeout
 *
 * @param func the function to be called
 * @param timeout optional timeout
 * @param funcArgs function arguments
 * @returns
 */
export function delayExecutionReturnClearTimeout(
  func: (...args: any[]) => void,
  timeout = 0,
  ...funcArgs: any[]
): () => void {
  const timer = setTimeout(() => func(...funcArgs), timeout);
  return () => {
    clearTimeout(timer);
  };
}
