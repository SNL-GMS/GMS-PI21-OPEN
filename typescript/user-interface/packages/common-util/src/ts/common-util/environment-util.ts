import type { CommonTypes } from '@gms/common-model';
import { UserProfileTypes } from '@gms/common-model';

import { isWindowDefined } from './window-util';

const windowIsDefined = isWindowDefined();

// !The environment utils currently depends on the common-model for UserMode
// TODO determine if the util library should only use primitive types
// TODO determine if the environment utils should be moved to a different package

/**
 * The NODE_ENV environment variable.
 */
export const { NODE_ENV } = process.env;

/**
 * True if NODE_ENV is set to development.
 */
export const IS_NODE_ENV_DEVELOPMENT = NODE_ENV === 'development';

/**
 * True if NODE_ENV is set to production.
 */
export const IS_NODE_ENV_PRODUCTION = NODE_ENV === 'production';

/**
 * True if NODE_ENV is set to test.
 */
export const IS_NODE_ENV_TEST = NODE_ENV === 'test';

/**
 * The UI Mode ['ian', 'soh' or 'legacy']
 */
export const GMS_UI_MODE = process.env.GMS_UI_MODE || undefined;

/**
 * True if configured for ALL; false otherwise.
 */
export const IS_MODE_LEGACY = GMS_UI_MODE === UserProfileTypes.UserMode.LEGACY;

/**
 * True if configured for IAN; false otherwise.
 */
export const IS_MODE_IAN = GMS_UI_MODE === UserProfileTypes.UserMode.IAN;

/**
 * True if configured for SOH; false otherwise.
 */
export const IS_MODE_SOH = GMS_UI_MODE === UserProfileTypes.UserMode.SOH;

/**
 * The current user mode, which defines which layouts are supported
 */
export const CURRENT_USER_MODE =
  (IS_MODE_LEGACY && UserProfileTypes.UserMode.LEGACY) ||
  (IS_MODE_IAN && UserProfileTypes.UserMode.IAN) ||
  (IS_MODE_SOH && UserProfileTypes.UserMode.SOH) ||
  '';

/** Useful because unit tests can easily override functions. */
export const isIanMode = (): boolean => IS_MODE_IAN;

/** Useful because unit tests can easily override functions. */
export const isSohMode = (): boolean => IS_MODE_SOH;

/**
 * Returns the supported modes based on the current user mode.
 */
export const SUPPORTED_MODES: UserProfileTypes.UserMode[] =
  (CURRENT_USER_MODE === UserProfileTypes.UserMode.SOH && [UserProfileTypes.UserMode.SOH]) ||
  (CURRENT_USER_MODE === UserProfileTypes.UserMode.IAN && [UserProfileTypes.UserMode.IAN]) ||
  Object.keys(UserProfileTypes.UserMode).map(mode => UserProfileTypes.UserMode[mode]);

/**
 * The UI_URL endpoint. This is the URL from which the UI content is served.
 */
export const UI_URL = windowIsDefined
  ? `${window.location.protocol}//${window.location.host}`
  : 'http://localhost:8080';

/**
 * The UI_BASE_PATH endpoint. This is the base path for the URL in a deployment
 */
export const UI_BASE_PATH =
  windowIsDefined && !window.location.host.includes('localhost') ? '/interactive-analysis-ui' : '';

/**
 * The SUBSCRIPTION protocol.
 */
export const UI_SUBSCRIPTION_PROTOCOL =
  windowIsDefined && window.location.protocol === 'https:' ? 'wss' : 'ws';

/**
 * The SUBSCRIPTION URL endpoint.
 */
export const UI_SUBSCRIPTION_URL = windowIsDefined
  ? `${UI_SUBSCRIPTION_PROTOCOL}://${window.location.host}`
  : 'ws://localhost';

/**
 * The GATEWAY_HTTP_PROXY_URI environment variable (or the default value if not set).
 */
export const GATEWAY_HTTP_PROXY_URI = process.env.GATEWAY_HTTP_PROXY_URI || UI_URL;

/**
 * The SUBSCRIPTIONS_PROXY_URI environment variable (or the default value if not set).
 */
export const SUBSCRIPTIONS_PROXY_URI = process.env.SUBSCRIPTIONS_PROXY_URI || UI_SUBSCRIPTION_URL;

/**
 * The API_GATEWAY_URI environment variable (or the default value if not set).
 */
export const API_GATEWAY_URI = GATEWAY_HTTP_PROXY_URI;

/**
 * The API_GATEWAY_URI environment variable for checking a user's login status.
 */
export const API_LOGIN_CHECK_URI = `${GATEWAY_HTTP_PROXY_URI}/interactive-analysis-api-gateway/auth/checkLogIn`;

/**
 * The API_GATEWAY_URI environment variable for accessing the login endpoint.
 */
export const API_LOGIN_URI = `${GATEWAY_HTTP_PROXY_URI}/interactive-analysis-api-gateway/auth/logInUser`;

/**
 * The API_GATEWAY_URI environment variable for accessing the logout endpoint.
 */
export const API_LOGOUT_URI = `${GATEWAY_HTTP_PROXY_URI}/interactive-analysis-api-gateway/auth/logOutUser`;

/**
 * The CESIUM_OFFLINE environment variable.
 */
export const CESIUM_OFFLINE = process.env.CESIUM_OFFLINE
  ? !(
      process.env.CESIUM_OFFLINE === 'null' ||
      process.env.CESIUM_OFFLINE === 'undefined' ||
      process.env.CESIUM_OFFLINE === 'false'
    )
  : false;

/**
 * The `AVAILABLE_SOUND_FILES` environment variable.
 * The available configured sound files for the system.
 */
export const AVAILABLE_SOUND_FILES: string[] =
  process.env.AVAILABLE_SOUND_FILES &&
  process.env.AVAILABLE_SOUND_FILES !== 'undefined' &&
  process.env.AVAILABLE_SOUND_FILES !== 'null'
    ? process.env.AVAILABLE_SOUND_FILES.split(';')
    : [];

export const VERSION_INFO: CommonTypes.VersionInfo = {
  versionNumber: process.env.VERSION_NUMBER ?? 'Unknown version',
  commitSHA: process.env.COMMIT_SHA ?? 'Unknown commit'
};

/**
 * Turns on timing points for UI. Set to 'verbose' to see log timing points.
 * Set to any other string to see warnings level timing only.
 */
export const GMS_PERFORMANCE_MONITORING_ENABLED =
  process.env.GMS_PERFORMANCE_MONITORING_ENABLED?.toLocaleLowerCase() ?? false;
