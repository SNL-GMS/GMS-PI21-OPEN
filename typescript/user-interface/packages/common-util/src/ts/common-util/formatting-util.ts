import { ValueType } from '../types/value-type';

/**
 * Rounds a number to the nearest specified decimal place; defaults to the nearest integer.
 *
 * @param value - the value to round
 * @param decimal - number of decimal spaces to round to; default is 0 which means it
 * rounds to the nearest integer
 *
 */
export const roundNumber = (value: number, decimal = 0): number => {
  const roundingHelper = 10;
  if (decimal < 0) {
    throw new Error(`Invalid decimal place provided for rounding`);
  }
  return Math.round((value * roundingHelper ** decimal) / roundingHelper ** decimal);
};

/**
 * Takes a number and sets it to a fixed precision,
 * also adds commas to delineate thousands places e.g. 1000 => 1,000
 *
 * @param value a value to be fixed
 * @param precision optional precision, default is based on config
 *
 * @returns a string
 */
export function setDecimalPrecision(value: number, precision = 2): string {
  if (value === undefined || value === null) return '';
  if (value.toFixed === undefined) {
    return '';
  }
  return value.toLocaleString(undefined, {
    maximumFractionDigits: precision,
    minimumFractionDigits: precision
  });
}

/**
 * Truncates a number to a certain decimal precision
 *
 * @param value
 * @param precision
 */
export const setDecimalPrecisionAsNumber = (value: number, precision = 2): number => {
  if (value === undefined || value === null || value.toFixed === undefined) return undefined;
  return parseFloat(value.toFixed(precision));
};

/**
 * Capitalizes the first character of each word
 *
 * @param input string input
 */
export function capitalizeFirstLetters(input: string): string {
  const words = input.split(' ');
  let newString = '';
  words.forEach(word => {
    if (word.length > 0) {
      const newWord = word[0].toUpperCase() + word.slice(1);
      newString += `${newWord} `;
    }
  });
  return newString.trim();
}

/**
 * Nicely capitalizes and removes _ from enums
 *
 * @param enumAsString string to prettify
 * @param stripFirstPart strips first part of enum IE (ENV_CLOCK_SLOW would be Clock Slow)
 *
 * @returns prettified string of enum
 */
export const prettifyAllCapsEnumType = (enumAsString: string, stripFirstPart = false): string => {
  if (enumAsString) {
    const asString: string = enumAsString;
    const regex = /_/g;
    const withoutUnderscores = asString.replace(regex, ' ');
    let stringArray: string[];
    if (stripFirstPart) {
      [, ...stringArray] = withoutUnderscores.toLowerCase().split(' ');
    } else {
      stringArray = withoutUnderscores.toLowerCase().split(' ');
    }
    return stringArray.map(s => String(s.charAt(0).toUpperCase()) + s.substring(1)).join(' ');
  }
  return undefined;
};

/**
 * Strips out everything up to the first occurrence of the `pattern` and removes first
 * occurrence of the pattern.
 *
 * @param str the string to strip
 * @param pattern the pattern to search for
 * @returns the stripped string
 */
export const stripOutFirstOccurrence = (str: string, pattern = '.'): string => {
  if (str && pattern) {
    if (str.includes(pattern)) {
      const [, ...channelNames] = str.split('.');
      return channelNames.join(pattern);
    }
  }
  return str;
};

/**
 * Determines Precision by type
 *
 * @param value the value
 * @param valueType value type
 * @param returnAsString determines return type
 * @returns a precision value as a string or number based on returnAsString
 */
export const determinePrecisionByType = (
  value: number,
  valueType: ValueType,
  returnAsString: boolean
): string | number => {
  switch (valueType) {
    case ValueType.PERCENTAGE:
      return returnAsString ? setDecimalPrecision(value, 2) : setDecimalPrecisionAsNumber(value, 2);
    case ValueType.INTEGER:
      return returnAsString ? setDecimalPrecision(value, 0) : setDecimalPrecisionAsNumber(value, 0);
    case ValueType.FLOAT:
      return returnAsString ? setDecimalPrecision(value, 2) : setDecimalPrecisionAsNumber(value, 2);
    default:
      return returnAsString ? setDecimalPrecision(value) : setDecimalPrecisionAsNumber(value);
  }
};
