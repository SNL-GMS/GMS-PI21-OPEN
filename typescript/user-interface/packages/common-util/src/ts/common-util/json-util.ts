/**
 * Converts an JS object to a prettified JSON object string.
 *
 * NOTE: This protects against and handles any circular references.
 *
 * @param object object to stringify and written to a file
 * @param fileName filename do NOT include extension
 */
export function jsonPretty(object: unknown): string {
  const getCircularReplacer = () => {
    const seen = new WeakSet();
    return (key, value) => {
      if (typeof value === 'object' && value !== null) {
        if (seen.has(value)) {
          return;
        }
        seen.add(value);
      }
      // eslint-disable-next-line consistent-return
      return value;
    };
  };
  return JSON.stringify(object, getCircularReplacer(), 2);
}
