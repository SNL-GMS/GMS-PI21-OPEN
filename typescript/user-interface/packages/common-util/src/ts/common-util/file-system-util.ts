import parse from 'csv-parse/lib/sync';
import fs from 'fs';

import { jsonPretty } from './json-util';
import { Logger } from './logger';

const logger = Logger.create('GMS_LOG_FILE_SYSTEM_UTIL', process.env.GMS_LOG_FILE_SYSTEM_UTIL);

/**
 * Reads the provided source JSON file into memory
 *
 * @param jsonFilePath The JSON filename from which to read the JSON content
 */
// !FIX: This should be generic and certainly should not always return an array
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function readJsonData<T = any>(jsonFilePath: string): T[] {
  const fileContents = fs.readFileSync(jsonFilePath, 'utf8');
  return JSON.parse(fileContents);
}

/**
 * Write the formatted JSON to file used by other write to file functions
 *
 * @param filePath
 * @param json
 */
export function writeJsonToFile(filePath: string, json: string): void {
  fs.writeFile(filePath, json, err => {
    if (err) {
      logger.warn(err.message);
    }
  });
}
/**
 * Writes provided object to file stringify and pretty
 *
 * @param object object to stringify and written to a file
 * @param fileName filename doNOT include extension
 */
export function writeJsonPretty(object: unknown, fileName: string): void {
  writeJsonToFile(`${fileName}.json`, jsonPretty(object));
}

/**
 * Saves given object to file
 * This will save us time from copying data from the terminal
 *
 * @param object object to save to file
 * @param filePath file path including file name
 */
export function writeObjectToJsonFile(object: unknown, filePath: string): void {
  writeJsonToFile(filePath, JSON.stringify(object, undefined, 2));
}

/** Utility functions for handling CSV files and other file related utils */

/**
 * Reads the provided source CSV file into memory
 *
 * @param csvFilePath
 */
// eslint-disable-next-line @typescript-eslint/no-explicit-any
export function readCsvData(csvFilePath: string): any[] {
  const fileContents = fs.readFileSync(csvFilePath, 'utf8');
  return parse(fileContents, { columns: true, delimiter: '\t' });
}

/**
 * Resolves the home value in the config and returns the path
 *
 * @param configString
 */
export function resolveHomeDataPath(configString: string): string[] {
  // Resolve the ${HOME} value or ${SOMETHING} value
  const regex = new RegExp(''.concat('\\$\\{', '([^}]+)', '\\}'), 'g');
  return [configString.replace(regex, (_, v) => process.env[v])];
}
