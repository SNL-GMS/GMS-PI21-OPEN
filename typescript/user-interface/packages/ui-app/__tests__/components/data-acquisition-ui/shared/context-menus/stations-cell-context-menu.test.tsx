import { ContextMenu } from '@blueprintjs/core';
import { Logger } from '@gms/common-util';
import React from 'react';
import { act } from 'react-dom/test-utils';

import type {
  StationSohContextMenuItemProps,
  StationSohContextMenuProps
} from '../../../../../src/ts/components/data-acquisition-ui/shared/context-menus/stations-cell-context-menu';
import {
  acknowledgeOnClick,
  DisabledStationSohContextMenu,
  getStationAcknowledgementMenuText,
  StationSohContextMenu
} from '../../../../../src/ts/components/data-acquisition-ui/shared/context-menus/stations-cell-context-menu';

const logger = Logger.create('GMS_LOG_JEST', process.env.GMS_LOG_JEST);

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
const Enzyme = require('enzyme');

describe('Station cell context menu', () => {
  const stationSohContextMenuProps: StationSohContextMenuProps = {
    stationNames: ['test', 'test2'],
    acknowledgeCallback: jest.fn()
  };

  const stationSohContextMenu = Enzyme.mount(
    // eslint-disable-next-line react/jsx-props-no-spreading
    <StationSohContextMenu {...stationSohContextMenuProps} />
  );
  const disabledStationSohContextMenu = Enzyme.mount(
    // eslint-disable-next-line react/jsx-props-no-spreading
    <DisabledStationSohContextMenu {...stationSohContextMenuProps} />
  );

  it('StationSohContextMenu can be created', () => {
    expect(stationSohContextMenu).toMatchSnapshot();
  });

  it('DisabledStationSohContextMenu can be created', () => {
    expect(disabledStationSohContextMenu).toMatchSnapshot();
  });
  it('getStationAcknowledgementMenuText to work as expected', () => {
    const stationNames = ['test', 'test1'];
    const withComment = false;
    const expectedResult = 'Acknowledge 2 stations';
    const result = getStationAcknowledgementMenuText(stationNames, withComment);
    expect(result).toEqual(expectedResult);
  });

  it('acknowledgeOnClick to work as expected', async () => {
    const stationSohContextMenuItemProps: StationSohContextMenuItemProps = {
      ...stationSohContextMenuProps,
      disabled: false
    };
    const createSpy = jest
      .spyOn(ContextMenu, 'show')
      .mockImplementation(() => logger.debug('shown'));
    const mouseEvent: any = {
      preventDefault: jest.fn(),
      clientX: 5,
      clientY: 5
    };
    // call acknowledge and wait for React lifecycle to finish so it can be deferred.
    await act(async () => {
      acknowledgeOnClick(mouseEvent, stationSohContextMenuItemProps);
      const waitDurationMs = 200;
      // eslint-disable-next-line no-promise-executor-return
      await new Promise(resolve => setTimeout(resolve, waitDurationMs));
    });
    expect(createSpy).toHaveBeenCalled();
  });
});
