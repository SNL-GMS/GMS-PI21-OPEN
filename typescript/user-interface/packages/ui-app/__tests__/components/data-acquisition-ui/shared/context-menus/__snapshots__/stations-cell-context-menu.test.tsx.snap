// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Station cell context menu DisabledStationSohContextMenu can be created 1`] = `
<DisabledStationSohContextMenu
  acknowledgeCallback={[MockFunction]}
  stationNames={
    Array [
      "test",
      "test2",
    ]
  }
>
  <Blueprint4.Menu>
    <ul
      className="bp4-menu"
    >
      <AcknowledgeMenuItem
        acknowledgeCallback={[MockFunction]}
        disabled={true}
        stationNames={
          Array [
            "test",
            "test2",
          ]
        }
      >
        <Blueprint4.MenuItem
          active={false}
          className="acknowledge-soh-status"
          data-cy="acknowledge-without-comment"
          disabled={true}
          multiline={false}
          onClick={[Function]}
          popoverProps={Object {}}
          selected={false}
          shouldDismissPopover={true}
          text="Acknowledge 2 stations"
        >
          <li
            className=""
          >
            <a
              className="bp4-menu-item bp4-disabled acknowledge-soh-status"
              data-cy="acknowledge-without-comment"
              tabIndex={-1}
            >
              <Blueprint4.Text
                className="bp4-fill"
                ellipsize={true}
              >
                <div
                  className="bp4-fill bp4-text-overflow-ellipsis"
                >
                  Acknowledge 2 stations
                </div>
              </Blueprint4.Text>
            </a>
          </li>
        </Blueprint4.MenuItem>
      </AcknowledgeMenuItem>
      <AcknowledgeWithCommentMenuItem
        acknowledgeCallback={[MockFunction]}
        disabled={true}
        stationNames={
          Array [
            "test",
            "test2",
          ]
        }
      >
        <Blueprint4.MenuItem
          active={false}
          className="acknowledge-soh-status"
          data-cy="acknowledge-with-comment"
          disabled={true}
          multiline={false}
          onClick={[Function]}
          popoverProps={Object {}}
          selected={false}
          shouldDismissPopover={true}
          text="Acknowledge 2 stations with comment..."
        >
          <li
            className=""
          >
            <a
              className="bp4-menu-item bp4-disabled acknowledge-soh-status"
              data-cy="acknowledge-with-comment"
              tabIndex={-1}
            >
              <Blueprint4.Text
                className="bp4-fill"
                ellipsize={true}
              >
                <div
                  className="bp4-fill bp4-text-overflow-ellipsis"
                >
                  Acknowledge 2 stations with comment...
                </div>
              </Blueprint4.Text>
            </a>
          </li>
        </Blueprint4.MenuItem>
      </AcknowledgeWithCommentMenuItem>
    </ul>
  </Blueprint4.Menu>
</DisabledStationSohContextMenu>
`;

exports[`Station cell context menu StationSohContextMenu can be created 1`] = `
<StationSohContextMenu
  acknowledgeCallback={[MockFunction]}
  stationNames={
    Array [
      "test",
      "test2",
    ]
  }
>
  <Blueprint4.Menu>
    <ul
      className="bp4-menu"
    >
      <AcknowledgeMenuItem
        acknowledgeCallback={[MockFunction]}
        disabled={false}
        stationNames={
          Array [
            "test",
            "test2",
          ]
        }
      >
        <Blueprint4.MenuItem
          active={false}
          className="acknowledge-soh-status"
          data-cy="acknowledge-without-comment"
          disabled={false}
          multiline={false}
          onClick={[Function]}
          popoverProps={Object {}}
          selected={false}
          shouldDismissPopover={true}
          text="Acknowledge 2 stations"
        >
          <li
            className=""
          >
            <a
              className="bp4-menu-item bp4-popover-dismiss acknowledge-soh-status"
              data-cy="acknowledge-without-comment"
              onClick={[Function]}
              tabIndex={0}
            >
              <Blueprint4.Text
                className="bp4-fill"
                ellipsize={true}
              >
                <div
                  className="bp4-fill bp4-text-overflow-ellipsis"
                >
                  Acknowledge 2 stations
                </div>
              </Blueprint4.Text>
            </a>
          </li>
        </Blueprint4.MenuItem>
      </AcknowledgeMenuItem>
      <AcknowledgeWithCommentMenuItem
        acknowledgeCallback={[MockFunction]}
        disabled={false}
        stationNames={
          Array [
            "test",
            "test2",
          ]
        }
      >
        <Blueprint4.MenuItem
          active={false}
          className="acknowledge-soh-status"
          data-cy="acknowledge-with-comment"
          disabled={false}
          multiline={false}
          onClick={[Function]}
          popoverProps={Object {}}
          selected={false}
          shouldDismissPopover={true}
          text="Acknowledge 2 stations with comment..."
        >
          <li
            className=""
          >
            <a
              className="bp4-menu-item bp4-popover-dismiss acknowledge-soh-status"
              data-cy="acknowledge-with-comment"
              onClick={[Function]}
              tabIndex={0}
            >
              <Blueprint4.Text
                className="bp4-fill"
                ellipsize={true}
              >
                <div
                  className="bp4-fill bp4-text-overflow-ellipsis"
                >
                  Acknowledge 2 stations with comment...
                </div>
              </Blueprint4.Text>
            </a>
          </li>
        </Blueprint4.MenuItem>
      </AcknowledgeWithCommentMenuItem>
    </ul>
  </Blueprint4.Menu>
</StationSohContextMenu>
`;
